source/libip6string/stoip6.c

```diff
@@ -69,11 +68,15 @@ bool stoip6(const char *ip6addr, size_t len, void *dest)
         addr = dest;
         memmove(addr + head_size + inserted_size, addr + head_size, tail_size);
         memset(addr + head_size, 0, inserted_size);
-    } else if (field_no != 8) {
-        //Report an error if we didn't get 8 fields
-        return false;
+    } else if (field_no != 8) { // Report an error if we didn't get 8 fields
+        goto error;
     }
     return true;
+
+error:
+    // Fill the output buffer with 0 so we stick to the old failure mechanism
```

Comment:
I don't think this exactly match the old failure in all cases (eg non-hex digits), so probably best to not claim that.

Reply:
Is the new comment less ambiguous or should I rather drop the mention of the old behavior ?